demos/object_detection_demo_ssd_async/main.cpp

```diff
@@ -50,523 +135,127 @@ bool ParseAndCheckCommandLine(int argc, char *argv[]) {
+
+    out.str("");
+    out << "Pool: " << std::fixed << std::setprecision(1) <<
+        info.numRequestsInUse << "/" << FLAGS_nireq;
+    putHighlightedText(frame, out.str(), cv::Point2f(10, 88), cv::FONT_HERSHEY_TRIPLEX, 0.6, cv::Scalar(200, 10, 10), 2);
 }
 
 int main(int argc, char *argv[]) {
     try {
         /** This demo covers certain topology and cannot be generalized for any object detection **/
-        std::cout << "InferenceEngine: " << *GetInferenceEngineVersion() << std::endl;
+        std::cout << "InferenceEngine: " << *InferenceEngine::GetInferenceEngineVersion() << std::endl;
 
         // ------------------------------ Parsing and validation of input args ---------------------------------
         if (!ParseAndCheckCommandLine(argc, argv)) {
             return 0;
         }
 
+        //------------------------------- Preparing Input ------------------------------------------------------
         slog::info << "Reading input" << slog::endl;
-        cv::VideoCapture cap;
-        if (!((FLAGS_i == "cam") ? cap.open(0) : cap.open(FLAGS_i.c_str()))) {
-            throw std::logic_error("Cannot open input file or camera: " + FLAGS_i);
-        }
-        const size_t width  = (size_t)cap.get(cv::CAP_PROP_FRAME_WIDTH);
-        const size_t height = (size_t)cap.get(cv::CAP_PROP_FRAME_HEIGHT);
-
-        // read input (video) frame
-        cv::Mat curr_frame;  cap >> curr_frame;
-        cv::Mat next_frame;
-
-        if (!cap.grab()) {
-            throw std::logic_error("This demo supports only video (or camera) inputs !!! "
-                                   "Failed getting next frame from the " + FLAGS_i);
-        }
-        // -----------------------------------------------------------------------------------------------------
+        auto cap = openImagesCapture(FLAGS_i, FLAGS_loop);
+        cv::Mat curr_frame;
 
-        // --------------------------- 1. Load inference engine ------------------------------------------------
-        slog::info << "Loading Inference Engine" << slog::endl;
-        Core ie;
-
-        slog::info << "Device info: " << slog::endl;
-        std::cout << ie.GetVersions(FLAGS_d);
+        //------------------------------ Running Detection routines ----------------------------------------------
+        std::vector<std::string> labels;
+        if (!FLAGS_labels.empty())
+            labels = DetectionModel::loadLabels(FLAGS_labels);
 
-        /** Load extensions for the plugin **/
-        if (!FLAGS_l.empty()) {
-            // CPU(MKLDNN) extensions are loaded as a shared library and passed as a pointer to base extension
-            IExtensionPtr extension_ptr = make_so_pointer<IExtension>(FLAGS_l.c_str());
-            ie.AddExtension(extension_ptr, "CPU");
+        ModelBase* model;
+        if (FLAGS_mt=="ssd")
+        {
+            model = new ModelSSD(FLAGS_m, (float)FLAGS_t, FLAGS_auto_resize, labels);
```

Comment:
You could pass `FLAGS_labels` and handle it in constructor.

Reply:
nope. This way we have an ability to pass manually created labels list (if we want to), or load them using static function.